admin/server/github.go

```diff
@@ -853,9 +972,294 @@ func (s *Server) fetchReposForInstallation(ctx context.Context, client *github.C
 	return repos, nil
 }
 
+func (s *Server) pushToGit(ctx context.Context, copyData func(dir, projPath string) error, repo, branch, subpath, token string, force bool) error {
+	ctx, cancel := context.WithTimeout(ctx, archivePullTimeout)
+	defer cancel()
+
+	// generate a temp dir to extract the archive
+	dir, err := os.MkdirTemp(os.TempDir(), "extracted_archives")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(dir)
+
+	// use a subfolder for working with git
+	gitPath := filepath.Join(dir, "proj")
```

Comment:
Can this be simplified using different directories for downloading archive and project directory ?
In that case we don't need to pass `dir` to `copyData` as well and it can just download it in any temp directory.

Reply:
This approach of extracting to the intended target avoids copy after extracting. With any temp directory we would have additional copy to the target directory (either root of the github repo or a selected subpath).

Reply:
May be I am missing something. 
1) For archive case the src is a zipped archive and destination is a project directory. We need to unzip the file and put it into its corresponding location in project directory. So it should be same whether src and destination are different directories or not.
2) For git case there is an explicit copy step from `tmp/extracted_archives/src` to `tmp/extracted_archives/proj`

2 got me thinking. For monorepos we seem to be copying `tmp/extracted_archives/src/subpath` to `tmp/extracted_archives/proj/subpath` (projPath) and then we do `git push` from this path. Will this work if rest of the contents especially `.git` at the root is not copied ?